pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/UnnecessaryFullyQualifiedNameRule.java

```diff
@@ -53,38 +57,44 @@ public Object visit(ASTClassOrInterfaceType node, Object data) {
     public Object visit(ASTName node, Object data) {
         if (!(node.jjtGetParent() instanceof ASTImportDeclaration)
                 && !(node.jjtGetParent() instanceof ASTPackageDeclaration)) {
+            // This name has no qualification, it can't be unnecessarily qualified
+            if (node.getImage().indexOf('.') < 0) {
+                return data;
+            }
             checkImports(node, data);
         }
         return data;
     }
 
+
+    /**
+     * Returns true if the name could be imported by this declaration.
+     * The name must be fully qualified, the import is either on-demand
+     * or static, that is its {@link ASTImportDeclaration#getImportedName()}
+     * is the enclosing package or type name of the imported type or static member.
+     */
+    private boolean declarationMatches(ASTImportDeclaration decl, String name) {
+        return name.startsWith(decl.getImportedName())
+                && name.lastIndexOf('.') == decl.getImportedName().length();
+    }
+
     private void checkImports(JavaNode node, Object data) {
         String name = node.getImage();
         matches.clear();
```

Comment:
I've seen, that matches is actually only used in this method, so I would remove the field and add it as a local variable here.

Reply:
Good catch